### PR TITLE
flowgger: init at 0.3.1

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes
a fast, simple and lightweight data collector written in Rust

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/flowgger/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  nativeBuildInputs = [
+    pkg-config
+    capnproto
+  ];
+
+  buildInputs = [ openssl ];
+
+  passthru.updateScript = ./update.sh;
+
+  # test failed
```
Comment (natsukium) [inline pkgs/tools/misc/flowgger/default.nix:37]: Why did tests fail?

Comment (ee2500) [reply pkgs/tools/misc/flowgger/default.nix:37]: This is the error message.

```
failures:

---- flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_multiple_sd stdout ----
thread 'flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_multiple_sd' panicked at 'assertion failed: `(left == right)`
  left: `"a:b\tc:123456\ta2:b2\tc2:123456\thost:testhostname\ttime:1691320524\tmessage:some test message\tfull_message:<23>Aug  6 11:15:24 testhostname appname[69]: 42 [someid a=\"b\" c=\"123456\"][someid2 a2=\"b2\" c2=\"123456\"] some test message\tlevel:7\tfacility:2\tappname:appname\tprocid:69\tmsgid:42"`,
 right: `"a:b\tc:123456\ta2:b2\tc2:123456\thost:testhostname\ttime:1659784524\tmessage:some test message\tfull_message:<23>Aug  6 11:15:24 testhostname appname[69]: 42 [someid a=\"b\" c=\"123456\"][someid2 a2=\"b2\" c2=\"123456\"] some test message\tlevel:7\tfacility:2\tappname:appname\tprocid:69\tmsgid:42"`', src/flowgger/encoder/ltsv_encoder.rs:196:5

---- flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_no_sd stdout ----
thread 'flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_no_sd' panicked at 'assertion failed: `(left == right)`
  left: `"host:testhostname\ttime:1691320524\tmessage:some test message\tfull_message:<23>Aug  6 11:15:24 testhostname appname[69]: 42 - some test message\tlevel:7\tfacility:2\tappname:appname\tprocid:69\tmsgid:42"`,
 right: `"host:testhostname\ttime:1659784524\tmessage:some test message\tfull_message:<23>Aug  6 11:15:24 testhostname appname[69]: 42 - some test message\tlevel:7\tfacility:2\tappname:appname\tprocid:69\tmsgid:42"`', src/flowgger/encoder/ltsv_encoder.rs:156:5


failures:
    flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_multiple_sd
    flowgger::encoder::ltsv_encoder::test_ltsv_full_encode_no_sd

test result: FAILED(B. 87 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.30s
```

Comment (ee2500) [reply pkgs/tools/misc/flowgger/default.nix:40]: I don't know what went wrong

Comment (natsukium) [reply pkgs/tools/misc/flowgger/default.nix:40]: I'm not sure what caused this error, but how about using `checkFlags` to skip it?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#disabling-package-tests-disabling-package-tests

Comment (ee2500) [reply pkgs/tools/misc/flowgger/default.nix]: Two tests were skipped.

**Review by emilylange — APPROVED** [review]

File: pkgs/tools/misc/flowgger/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+, openssl
+, capnproto
+, CoreServices
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "flowgger";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
```
Comment (natsukium) [inline pkgs/tools/misc/flowgger/default.nix:15]: How about using `fetchCrate` here?
I don't think we need to maintain Cargo.lock ourselves with using it.

Comment (ee2500) [reply pkgs/tools/misc/flowgger/default.nix]: It is a good idea. I changed source file

**Review by drupol — APPROVED** [review]

**Pull request merged**